controller/status.go

```diff
@@ -16,6 +16,8 @@ func (usl UnitStatusList) Group() ([]fleet.UnitStatus, error) {
 	matchers := map[string]struct{}{}
 	newList := []fleet.UnitStatus{}
 
+	hashesEqual := allHashesEqual(usl)
```

Comment:
This is the trick. When we added unit hashes we compared the unit hashes of the unit files that were already grouped. Obviously it makes no sense to compare unit hashes with the units within the same group. We are interested in comparing the unit hashes across the slices.